vignettes/easier_manual.Rmd

```diff
@@ -18,123 +18,218 @@ knitr::opts_chunk$set(
-patient_labels <- Riaz_data$patient_response
+For the first scenario, we can use the patients' response information available from Mariathasan cohort. Because patients' tumor mutational burden is also provided, we can also used to compare its performance against the quantitative descriptors. 
 
-# *********************
-# Specify output file path
-output_file_path <- "/Users/Oscar/Desktop/Riaz"
+Additionally the user can set `easier_with_TMB = TRUE` to apply a refined approach based on the integration of `easier` predictions with information on tumor mutational burden (TMB). 
+Since both immune response and TMB are essential for an effective immunotherapy response, we decided to conceptualize this in our predictions by either penalizing or weighting differently our scores in high- and low-TMB patients. 
+
+```{r eval = FALSE}
+# assess easier predicted immune response
+assess_immune_response(predictions_immune_response = predictions_immune_response,
+                       real_patient_response = obj_dataset@response,
+                       RNA_tpm = obj_dataset@tpm,
+                       output_file_path = "../figures",
+                       cancer_type = obj_dataset@cancertype,
+                       TMB_values = obj_dataset@TMB,
+                       easier_with_TMB = TRUE)
 
-compare_immune_response(predictions_immune_response = predictions_immune_response,
-                        real_patient_response = patient_labels,
-                        RNA.tpm = tpm,
-                        output_file_path = output_file_path,
-                        cancertype = cancertype)
+```
 
+We can go further and analyze systems biomarkers present in the user's bulk RNA-seq data. `explore_biomarkers` allows to investigate mechanisms behind patients' response to treatment through interpretable biomarkers. 
 
-# ---------------------------------------------------------------------------
-# Have a look into the biomarkers
-# ---------------------------------------------------------------------------
+In order to leverage the biomarkers information obtained during model training, you need to specify again which `cancer_type` the bulk RNA-seq data belongs.
 
-explore_biomarkers(pathways = pathways_scores,
+```{r eval = FALSE}
+explore_biomarkers(pathways = pathway_activity,
                    immunecells = cell_fractions,
-                   # lrpairs = lrpairs_weights,
-                   # tfs = tfs_scores,
-                   # ccpairsgroupedscores = ccpairsgrouped_scores,
-                   cancertype = cancertype,
-                   real_patient_response = patient_labels,
-                   output_file_path = "/Users/Oscar/Desktop/Riaz")
+                   lrpairs = lrpair_weights,
+                   tfs = tf_activity,
+                   ccpairs = ccpair_scores,
+                   cancer_type = obj_dataset@cancertype,
+                   real_patient_response = obj_dataset@response,
```

Comment:
Here's an example of when we use "accessors".

If we want to keep the class, we would ideally just need something like 

```
get_cancertype(obj_dataset)
```

that does nothing but accessing the slot like we do, but not in a way that is exposed to the users